index.js

```diff
@@ -1,73 +1,102 @@
+    testCases: [
+      {
         input: "[1,2,3,4,5]",
-        output: "5"
-    }]
-}];
-
-
-const SUBMISSION = [
+        output: "5",
+      },
+    ],
+  },
+];
 
-]
+const SUBMISSION = [];
 
-app.post('/signup', function(req, res) {
+app.post("/signup", function (req, res) {
   // Add logic to decode body
-  // body should have email and password
 
+  // body should have email and passwor
+  const { email, password } = req.body;
 
   //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
+  const existinguser = USERS.filter((user) => user.email == email);
+
+  if (existinguser.length) {
+    res.status(400).json({ message: "user already exists" });
+  }
 
+  //Add user to Array
+  USERS.push({ email, password });
 
   // return back 200 status code to the client
-  res.send('Hello World!')
-})
+  res.status(200).json({ message: "user added successfully" });
+});
 
-app.post('/login', function(req, res) {
+app.post("/login", function (req, res) {
   // Add logic to decode body
   // body should have email and password
 
+  const { email, password } = req.body;
+
   // Check if the user with the given email exists in the USERS array
   // Also ensure that the password is the same
-
+  const existinguser = USERS.filter((user) => user.email == email);
 
   // If the password is the same, return back 200 status code to the client
   // Also send back a token (any random string will do for now)
   // If the password is not the same, return back 401 status code to the client
+  if (existinguser.password === password) {
+    res.status(200).json({ message: "You are logged in ", token: "mytoken" });
+  } else {
+    res
+  .status(401)
+      .json({ message: "Some thing goes wrong check email/password" });
+  }
+
+  res.send("Hello World from route 2!");
+});
 
-
-  res.send('Hello World from route 2!')
-})
-
-app.get('/questions', function(req, res) {
-
+app.get("/questions", function (req, res) {
   //return the user all the questions in the QUESTIONS array
-  res.send("Hello World from route 3!")
-})
-
-app.get("/submissions", function(req, res) {
-   // return the users submissions for this problem
-  res.send("Hello World from route 4!")
+  res.json(QUESTIONS);
 });
 
+app.get("/submissions", function (req, res) {
+  // return the users submissions for this problem
+  res.json({ message: "submitted the question" });
+});
 
-app.post("/submissions", function(req, res) {
-   // let the user submit a problem, randomly accept or reject the solution
-   // Store the submission in the SUBMISSION array above
-  res.send("Hello World from route 4!")
+app.post("/submissions", function (req, res) {
+  // let the user submit a problem, randomly accept or reject the solution
+  const question = req.body;
+  const checkvalue = Math.floor(Math.random()*10);
+ console.log(checkvalue)
+  if(checkvalue<=3){
+    res.json({message:"Sorry,Try Again"})
+  }else{
+    SUBMISSION.push(question)
```

Comment:
In General any website similar to leetcode will only let logged in users to submit solutions.

it's good to verify if the user exists in your datastore. storing submissions without checking identity of users will let anybody to post a submission without creating the account.

attaching user identity with submissions will also unlock the feature to list particular users submission history